src/index.js

```diff
@@ -69,15 +96,34 @@ export default function ({ types: t }) {
           node.object = t.identifier(opts.systemGlobal);
         }
       },
-      Identifier({ node }) {
+      ReferencedIdentifier(path, state) {
+        if (path.node.name == 'define' &&
+          !path.scope.hasBinding('define') &&
+          (!t.isExpression(path.parentPath) ||
+            (t.isUnaryExpression(path.parentPath) && path.parentPath.node.operator === 'typeof'))) {
+          path.replaceWith(t.identifier('undefined'));
+        }
+      },
+      Identifier(path) {
+        let { node } = path;
         // test if file paths are used
         if (t.isIdentifier(node, { name: '__filename' }) ||
           t.isIdentifier(node, { name: '__dirname' })) {
           this.usesFilePaths = true;
         }
+
+        if (t.isIdentifier(node, { name: 'module' }) &&
+         t.isVariableDeclarator(path.parent)) {
+          this.moduleRedeclaredInOuterScope = true;
+        }
```

Comment:
Yes, that should be fixed if we decide to leave the module.export return statement in place. But as @guybeford already mentioned, it would perhaps make sense to completely remove the return statement? Then this test would be obsolete.